<launch>
    <!-- Launch script for Multi Agent sim with suda March 2019 -->
    <arg name="only_command" default="true" />
    <arg name="is_multicopter" default="true" />


<!-- ______________________________ UAV 1 ____________________________________ -->

    <group ns="uav1">
        <node pkg="offboard_control" name="Quad1_offboard" type="offboard_control_node" output="screen">
            <param name ="is_multicopter" value= "$(arg is_multicopter)" />
            <param name="att_mode_rate" value="40"/>
            <param name="pva_mode_rate" value="20"/>
            <param name="landing_z" value="0.11"/>
            <param name="takeoff_z" value="2.0"/>
            <param name="takeoff_Kp" value="1.0" />
            <param name="landing_Kp" value="1.0" />

            <!-- Position Control parameters -->
            <param name="mass" value="1.535"/>       <!-- kg -->
            <param name="gz" value="9.81"/>          <!-- m/s^2 -->
            <param name="thrustRatio" value="1.7"/>  <!-- maxThrust/Weight -->
            <param name="kpx" value="10"/>
            <param name="kpy" value="10"/>
            <param name="kpz" value="10"/>
            <param name="kvx" value="5"/>
            <param name="kvy" value="5"/>
            <param name="kvz" value="7.5"/>
            <param name="kix" value="0"/>
            <param name="kiy" value="0"/>
            <param name="kiz" value="0"/>
            <param name="maxInteg_x" value="0"/>
            <param name="maxInteg_y" value="0"/>
            <param name="maxInteg_z" value="0"/>
        </node>
    </group>

    <node pkg="offboard_control" name="Quad1_action" type="multi_agent_suda_uav_flight_02.py" output="screen">
        <param name="quad_ros_namespace" value="/uav1" />
        <param name="xOffset" value="42.5" />
        <param name="yOffset" value="25.5" />
        <param name="zOffset" value="0" />
        <param name="base_x" value="5" />
        <param name="base_y" value="5" />
        <param name="max_x" value="45" />
        <param name="max_y" value="45" />
        <param name="nb_y" value="40" />
        <param name="nb_x" value="40" />
        <param name="freq" value="20" />
        <param name="time_full_traj" value="3" />
        <param name="Abs_file_path" value="/home/jaq394l/catkin_ws/src/PX4_ROS_packages/offboard_control/src/MultiAgentSuda/" />
        <param name="controller_file" value="Controller_Safety_01_01.json" />
        <param name="Allowed_states_pickle_file" value="allowedstates_01.json" />
        <param name="Invis_states_pickle_file" value="iset_01.json" />
        <param name="outside_region_state" value="1600" />
    </node>

<!-- ______________________________ UAV 2 ____________________________________ -->

    <group ns="uav2">
        <node pkg="offboard_control" name="Quad2_offboard" type="offboard_control_node" output="screen">
            <param name ="is_multicopter" value= "$(arg is_multicopter)" />
            <param name="att_mode_rate" value="40"/>
            <param name="pva_mode_rate" value="20"/>
            <param name="landing_z" value="0.11"/>
            <param name="takeoff_z" value="2.0"/>
            <param name="takeoff_Kp" value="1.0" />
            <param name="landing_Kp" value="1.0" />

            <!-- Position Control parameters -->
            <param name="mass" value="1.535"/>       <!-- kg -->
            <param name="gz" value="9.81"/>          <!-- m/s^2 -->
            <param name="thrustRatio" value="1.7"/>  <!-- maxThrust/Weight -->
            <param name="kpx" value="10"/>
            <param name="kpy" value="10"/>
            <param name="kpz" value="10"/>
            <param name="kvx" value="5"/>
            <param name="kvy" value="5"/>
            <param name="kvz" value="7.5"/>
            <param name="kix" value="0"/>
            <param name="kiy" value="0"/>
            <param name="kiz" value="0"/>
            <param name="maxInteg_x" value="0"/>
            <param name="maxInteg_y" value="0"/>
            <param name="maxInteg_z" value="0"/>
        </node>
    </group>

    <node pkg="offboard_control" name="Quad2_action" type="multi_agent_suda_uav_flight_02.py" output="screen">
        <param name="quad_ros_namespace" value="/uav2" />
        <param name="xOffset" value="22.5" />
        <param name="yOffset" value="25.5" />
        <param name="zOffset" value="0" />
        <param name="base_x" value="5" />
        <param name="base_y" value="5" />
        <param name="max_x" value="45" />
        <param name="max_y" value="45" />
        <param name="nb_y" value="40" />
        <param name="nb_x" value="40" />
        <param name="freq" value="20" />
        <param name="time_full_traj" value="3" />
        <param name="Abs_file_path" value="/home/jaq394l/catkin_ws/src/PX4_ROS_packages/offboard_control/src/MultiAgentSuda/" />
        <param name="controller_file" value="Controller_Safety_02_01.json" />
        <param name="Allowed_states_pickle_file" value="allowedstates_01.json" />
        <param name="Invis_states_pickle_file" value="iset_01.json" />
        <param name="outside_region_state" value="1600" />
    </node>

<!-- ______________________________ Other Nodes ____________________________________ -->

    <node pkg="min_snap_ros" name="min_snap" type="min_snap_ros_node" output="screen"/>

</launch>
